cibuildwheel/macos.py

```diff
@@ -14,10 +14,10 @@
 def get_python_configurations(build_selector):
     PythonConfiguration = namedtuple('PythonConfiguration', ['version', 'identifier', 'url'])
     python_configurations = [
-        PythonConfiguration(version='2.7', identifier='cp27-macosx_10_6_intel', url='https://www.python.org/ftp/python/2.7.17/python-2.7.17-macosx10.6.pkg'),
-        PythonConfiguration(version='3.5', identifier='cp35-macosx_10_6_intel', url='https://www.python.org/ftp/python/3.5.4/python-3.5.4-macosx10.6.pkg'),
-        PythonConfiguration(version='3.6', identifier='cp36-macosx_10_6_intel', url='https://www.python.org/ftp/python/3.6.8/python-3.6.8-macosx10.6.pkg'),
-        PythonConfiguration(version='3.7', identifier='cp37-macosx_10_6_intel', url='https://www.python.org/ftp/python/3.7.6/python-3.7.6-macosx10.6.pkg'),
+        PythonConfiguration(version='2.7', identifier='cp27-macosx_10_9_x86_64', url='https://www.python.org/ftp/python/2.7.17/python-2.7.17-macosx10.9.pkg'),
+        PythonConfiguration(version='3.5', identifier='cp35-macosx_10_9_x86_64', url='https://www.python.org/ftp/python/3.5.4/python-3.5.4-macosx10.6.pkg'),
+        PythonConfiguration(version='3.6', identifier='cp36-macosx_10_9_x86_64', url='https://www.python.org/ftp/python/3.6.8/python-3.6.8-macosx10.9.pkg'),
+        PythonConfiguration(version='3.7', identifier='cp37-macosx_10_9_x86_64', url='https://www.python.org/ftp/python/3.7.6/python-3.7.6-macosx10.9.pkg'),
```

Comment:
Aha, so we have a little snag here - the build identifiers are changing - both the machine and the macos system. 

Previously I've mused about renaming these e.g. `cp37-macosx_10_9_x86_64` →  `cp37-macosx`. Thinking about this fresh...

The architecture isn't needed any longer, since we only build x86_64. Should we remove it? Will we ever need an architecture switch on macos again? Hm, well, [ARM Macs might exist](https://www.theinquirer.net/inquirer/news/3082867/apple-arm-macbook-2020-bloomberg) one day! And, all other platforms contain an architecture flag, so keeping it retains consistency.

Then there's the `10_9` bit - we expect to remove this part because #156 should be making this conditional on the MACOSX_DEPLOYMENT_TARGET var instead.

So here's what I'm thinking... rather than making users update their settings from `cp37-macosx_10_6_intel` to `cp37-macosx_10_9_x86_64` and then from `cp37-macosx_10_9_x86_64` to `cp37-macosx_x86_64` when #156 lands, let's pull in the [build selector migration](https://github.com/joerick/cibuildwheel/pull/156/files#diff-dfdf4eb7a2b7f84841e9d286a4063e03R204-R208) and go to the `cp37-macosx_x86_64` format in this PR.

Thoughts @YannickJadoul @Czaki ?

Reply:
Depends a bit on when we can merge #156, i.e., when a new wheel version will be released.

If that's before you plan a new release, I'd argue to keep this as it is now and not mix different PRs. If wheel won't release too soon, then yeah, maybe. But maybe that's still another clean PR, then?
(I've asked, btw: https://github.com/pypa/wheel/pull/314#issuecomment-569087959; let's see what comes out of that.)

On keeping the architecture: I don't think that's really bothering? It even keeps things clear, that wheels will only be built for 64-bit, I'd say. And with the wildcard matching on `*-macosx_*`, most users might not even need to change too much (I'd hope users don't depend on this 10_6/10_9 too much?). And indeed: keeping it would be good for the sake of consistency with `manylinux_x86_64`/`manylinux_i686`/ `win32`/`win_amd64`, where also now always have architecture, but not the platform's version :-)

Reply:
Afterthought: is there any reason to use the `10.9` packages from Python.org? We should be able to build 64-bit-only wheels that are still 10.6-compatible, or not, @mayeut? In that case, we can still keep our old build identifiers for a little while, until #156.

Reply:
Thanks for getting in touch with the wheel folks! I had been trying to unblock a few PRs but hopefully that's moving now.

> Afterthought: is there any reason to use the 10.9 packages from Python.org?

I _think_ this is more convenient because the 10.9 packages build a x86_64 wheel by default, but I'll let @mayeut correct me if I'm wrong...